src/coreclr/gc/gc.cpp

```diff
@@ -4964,6 +4964,25 @@ BOOL gc_heap::reserve_initial_memory (size_t normal_size, size_t large_size, siz
         }
     }
 
+    if (reserve_success && separated_poh_p)
+    {
+        for (int heap_no = 0; reserve_success && heap_no < num_heaps; heap_no++)
```

Comment:
nit
`for (int heap_no = 0; reserve_success && heap_no < num_heaps; heap_no++)`
->
`for (int heap_no = 0; (reserve_success && (heap_no < num_heaps)); heap_no++)`